### PR TITLE
[CFP-70] FormBuilder migration - Transfer fee

### PR DESCRIPTION
**Pull request by NenehB**
What

Continue the migration away from the deprecated design system to the current GOVUK Frontend.

Ticket

[FormBuilder migration -  Transfer fee](https://dsdmoj.atlassian.net/browse/CFP-70)

Why

- Update the service to use GOV.UK styles, components and patterns
- To improve and resolve the service's accessibility issues
- To reduce code complexity
- To improve maintainability

--------

#### TODO (wip)

 - [x] Fix test
 - [x] Update spec
 - [x] Update aob

### CODE REVIEW

File: app/views/external_users/claims/transfer_fee/_fields.html.haml [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
       - unless @claim.transfer_detail.elected_case?
 
         - if @claim.transfer_detail.days_claimable?
           .js-fee-calculator-effector
-            = f.adp_text_field :actual_trial_length,
-              label: t('.actual_trial_length'),
-              input_classes:'form-control form-control-1-4 js-fee-calculator-days',
-              input_type: 'number',
-              errors: @error_presenter
+            = f.govuk_number_field :actual_trial_length,
```
Comment (imtiazAhmd) [inline app/views/external_users/claims/transfer_fee/_fields.html.haml]: Parent div with class `.js-fee-calculator-effector` can be replaced with the parameter `form_group: { classes: 'js-fee-calculator-effector' }`

File: app/views/external_users/claims/transfer_fee/_fields.html.haml [hunk 2]
```diff
@@ -1,42 +1,47 @@
...
+              width: 'one-quarter'
 
         .js-fee-calculator-effector
-          = tf.adp_text_field :quantity,
-            label: t('.quantity'),
-            input_type: 'number',
-            input_classes:'quantity fee-quantity js-fee-calculator-ppe form-control-1-4',
+          = tf.govuk_number_field :quantity,
+            class: 'quantity fee-quantity js-fee-calculator-ppe',
+            label: { text: t('.quantity') },
```
Comment (imtiazAhmd) [inline app/views/external_users/claims/transfer_fee/_fields.html.haml]: Parent div with class `.js-fee-calculator-effector` can be replaced with the parameter `form_group: { classes: 'js-fee-calculator-effector' }`

File: app/views/external_users/claims/transfer_fee/_fields.html.haml [hunk 3]
```diff
@@ -1,42 +1,47 @@
...
-        = tf.adp_text_field :amount,
-            label: t('.amount'),
-            input_classes:'total fee-amount form-input-denote__input form-control-1-4 ',
-            input_type: 'currency',
-            value: number_with_precision(tf.object.amount, precision: 2),
-            error_key: 'transfer_fee.amount',
-            errors: @error_presenter
+        = tf.govuk_number_field :amount,
+          class: 'total fee-amount',
+          label: { text: t('.amount') },
```
Comment (imtiazAhmd) [inline app/views/external_users/claims/transfer_fee/_fields.html.haml]: Parent div with class `.js-fee-calculator-effector > .calculated-grad-fee` can be replaced with the parameter `form_group: { classes: 'calculated-grad-fee js-fee-calculator-effector' }` 
as JS only looking for any element in the class  `js-graduated-price-effectee`

File: app/views/external_users/claims/transfer_fee/_fields.html.haml [hunk 4]
```diff
@@ -1,42 +1,47 @@
...
+          min: 0,
+          prefix_text: '£',
+          value: number_with_precision(tf.object.amount, precision: 2),
+          width: 'one-quarter'
 
       .js-fee-calculator-success
-        = tf.hidden_field :price_calculated, value: tf.object.price_calculated?
+        = tf.govuk_text_field :price_calculated,
```
Comment (imtiazAhmd) [inline app/views/external_users/claims/transfer_fee/_fields.html.haml]: Parent div with class `.js-fee-calculator-success` can be replaced with the parameter `form_group: { classes: 'js-fee-calculator-success' }`

File: app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js [hunk 5]
```diff
@@ -107,7 +107,7 @@
 
     setAmount: function (data, context) {
       const $amount = $(context).find('input.fee-amount')
-      const $priceCalculated = $(context).find('.js-fee-calculator-success > input')
+      const $priceCalculated = $(context).find('.js-fee-calculator-success input')
```
Comment (imtiazAhmd) [inline app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.GraduatedPrice.js]: is this change necessary ? looking at the code it seems like `input` is directly under the parent div with class `js-fee-calculator-success`. 
Hence using a child selector `>` is more appropriate as this will only target the immediate child elements and not any descendants.

File: config/locales/en/error_messages/claim.yml [hunk 6]
```diff
@@ -1387,27 +1387,27 @@ transfer_fee:
...
-      short: *enter_valid_quantity
+      short: _
       api: Enter a valid PPE quantity for the transfer fee
-    item_max_amount:
+    enter_a_valid_ppe_quantity_for_the_transfer_fee:
       long: Enter a valid PPE quantity for the transfer fee
-      short: *enter_valid_quantity
+      short: _
       api: Enter a valid PPE quantity for the transfer fee
   amount:
```
Comment (naseberry) [inline config/locales/en/error_messages/claim.yml:1401]: Should `amount` have a key:value pair for the attribute `:item_max_amount`
The validator `def validate_presence_and_numericality_govuk_formbuilder` validates against `:blank`, `:numericality` and `:item_max_amount`

**Review by imtiazAhmd — APPROVED** [review]
LGTM 👍🏼

**Review by naseberry — APPROVED** [review]
💯
